src/mmw/js/src/modeling/views.js

```diff
@@ -131,6 +133,7 @@ var ProjectMenuView = Marionette.ItemView.extend({
 
         rename.on('update', function(val) {
             self.model.updateName(val);
+            self.model.saveProjectAndScenarios();
```

Comment:
Removing the listener on the model and moving the call here might introduce problematic edge cases. I haven't seen any but its just something to be cautious about.